src/test/java/com/chat/yourway/unit/service/impl/ContactServiceImplTest.java

```diff
@@ -80,4 +103,115 @@ public void changePassword_shouldThrowOldPasswordsIsNotEqualToNewException_whenU
         verify(passwordEncoder, never()).encode(anyString());
         verify(contactRepository, never()).changePasswordByUsername(anyString(), anyString());
     }
+
+    @Test
+    @DisplayName("send email to restore password should send email when user account exists by email")
+    public void sendEmailToRestorePassword_shouldSendEmail_whenUserAccountExistsByEmail() {
+        // Given
+        var username = "username12353";
+        var email = "user@gmail.com";
+        var emailMessageType = EmailMessageType.RESTORE_PASSWORD;
+        var path = "path";
+        var contact = Contact.builder()
+                .id(1)
+                .username(username)
+                .email(email)
+                .password("123456")
+                .isActive(true)
+                .isPrivate(true)
+                .build();
+        var emailMessage = new EmailMessageDto(email, emailMessageType.getSubject(), emailMessageType.getMessageBody());
+
+        // When
```

Comment:
When section is responsible for calling processing. Given is responsible for setup. So move all when/doNothings to "Given"